### PR TITLE
Kit: Fix - Update site settings do not updates kit site identity.

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: core/kits/documents/kit.php [hunk 1]
```diff
@@ -14,6 +14,14 @@
 
 class Kit extends PageBase {
 
+	public static $site_options_to_kit_dictionary = [
+		'blogname' => 'settings.site_name',
```
Comment (matipojo) [inline core/kits/documents/kit.php:18]: Move to `tabs/settings-site-identity.php`.

Use a simple format: 
```
'site_name' => [ 
	'type' => 'option',
	'name' => 'blogname',
...
```

File: core/kits/documents/tabs/settings-site-identity.php [hunk 2]
```diff
@@ -93,23 +94,34 @@ protected function register_tab_controls() {
...
+			$this->update_option_based_on_kit( $option_name, $kit_path, $data );
 		}
 
 		if ( isset( $data['settings']['site_logo'] ) ) {
 			set_theme_mod( 'custom_logo', $data['settings']['site_logo']['id'] );
 		}
+	}
+
+	protected function update_option_based_on_kit( $option_name, $kit_path, $data ) {
+		$value = $data;
```
Comment (matipojo) [inline core/kits/documents/tabs/settings-site-identity.php:115]:
```suggestion
		$value = '';
```

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/base/document.php [hunk 3]
```diff
@@ -49,6 +49,11 @@ abstract class Document extends Controls_Stack {
 	 */
 	protected $post;
 
+	/**
+	 * @var bool
+	 */
+	protected $saving = false;
```
Comment (matipojo) [inline core/base/document.php:55]:
```suggestion
	protected $is_saving = false;
```

File: core/kits/documents/kit.php [hunk 4]
```diff
@@ -78,9 +77,13 @@ public function save( $data ) {
 		$saved = parent::save( $data );
 
 		if ( $saved ) {
+			$this->set_saving( true );
```
Comment (matipojo) [inline core/kits/documents/kit.php:80]: Explain.

File: core/kits/documents/tabs/settings-site-identity.php [hunk 5]
```diff
@@ -93,9 +93,14 @@ protected function register_tab_controls() {
 	}
 
 	public function on_save( $data ) {
-		if ( ! isset( $data['settings'] ) || DB::STATUS_PUBLISH !== $data['settings']['post_status'] ) {
+		if (
+			! isset( $data['settings'] ) ||
+			DB::STATUS_PUBLISH !== $data['settings']['post_status'] ||
+			strpos( current_action(), 'update_option_' ) === 0
```
Comment (matipojo) [inline core/kits/documents/tabs/settings-site-identity.php:99]: Explain.

File: core/kits/manager.php [hunk 6]
```diff
@@ -177,6 +177,17 @@ public function get_kit_for_frontend() {
...
 
+	public function update_kit_settings_based_on_option( $key, $value ) {
+		/** @var Kit $active_kit */
+		$active_kit = $this->get_active_kit();
+
+		if ( $active_kit->is_saving() ) {
+			return;
+		}
+
+		$active_kit->save( [ 'settings' => [ $key => $value ] ] );
```
Comment (matipojo) [inline core/kits/manager.php:188]: Please try update blogname+description and check how many revisions are created (check post_parent=kit_id)

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/kits/manager.php (hunk 6 above)
Comment (Nevoss) [reply core/kits/manager.php:188]: It does not create revisions at all.

File: tests/phpunit/elementor/core/kits/test-manager.php [hunk 7]
```diff
@@ -5,7 +5,7 @@
 use Elementor\Plugin;
 use Elementor\Testing\Elementor_Test_Base;
 
-class Test_Upgrades extends Elementor_Test_Base {
+class Elementor_Test_Manager extends Elementor_Test_Base {
```
Comment (matipojo) [inline tests/phpunit/elementor/core/kits/test-manager.php:8]:
```suggestion
class Test_Manager extends Elementor_Test_Base {
```

File: tests/phpunit/elementor/core/kits/test-manager.php [hunk 8]
```diff
@@ -38,4 +38,26 @@ public function test_get_active_id() {
 		$active_id_after_invalidate = Plugin::$instance->kits_manager->get_active_id();
 		$this->assertNotEquals( $active_id, $active_id_after_invalidate, $test_description );
 	}
+
+	public function test_when_updating_blogname_or_blogdescription_option_it_should_update_the_kit(  ) {
+		$name = get_option('blogname');
+		$description = get_option('blogdescription');
+
+		$expectedName = 'My new name';
```
Comment (matipojo) [inline tests/phpunit/elementor/core/kits/test-manager.php:46]:
```suggestion
		$expectedName = 'Test name';
```

File: tests/phpunit/elementor/core/kits/test-manager.php [hunk 9]
```diff
@@ -38,4 +38,26 @@ public function test_get_active_id() {
...
+		$expectedDescription = 'My new description';
```
Comment (matipojo) [inline tests/phpunit/elementor/core/kits/test-manager.php:47]:
```suggestion
		$expectedDescription = 'Test description';
```

File: tests/phpunit/elementor/core/kits/test-manager.php [hunk 10]
```diff
@@ -38,4 +38,26 @@ public function test_get_active_id() {
...
+		update_option('blogname', $expectedName);
+		update_option('blogdescription', $expectedDescription);
+
+		$kit = Plugin::$instance->kits_manager->get_active_kit();
+
+		$this->assertEquals($expectedName, $kit->get_settings('site_name'));
+		$this->assertEquals($expectedDescription, $kit->get_settings('site_description'));
+
+		update_option('blogname', $name);
+		update_option('blogdescription', $description);
```
Comment (matipojo) [inline tests/phpunit/elementor/core/kits/test-manager.php:61]: PHPCS.

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — DISMISSED** [review]

File: core/base/document.php [hunk 11]
```diff
@@ -49,6 +49,11 @@ abstract class Document extends Controls_Stack {
 	 */
 	protected $post;
 
+	/**
+	 * @var bool
+	 */
+	protected $is_saving = false;
```
Comment (kobizz) [inline core/base/document.php:55]: Since you have a setter, it's better to make it private

**Review by kobizz — CHANGES_REQUESTED** [review]

**Review by matipojo — APPROVED** [review]

**Review by kobizz — APPROVED** [review]

**Pull request merged**